driver/src/models/tokens.rs

```diff
@@ -0,0 +1,94 @@
+//! Module implements common data types for tokens on the exchange.
+
+use anyhow::{Context, Error, Result};
+use serde::de::Error as _;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+/// A token ID wrapper type that implements JSON serialization in the solver
+/// format.
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialOrd, PartialEq)]
+pub struct TokenId(pub u16);
+
+impl<'de> Deserialize<'de> for TokenId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let key = Cow::<str>::deserialize(deserializer)?;
+        if !key.starts_with('T') || key.len() != 5 {
+            return Err(D::Error::custom("Token ID must be of the form 'Txxxx'"));
+        }
+
+        let id = key[1..].parse::<u16>().map_err(D::Error::custom)?;
+        Ok(TokenId(id))
+    }
+}
+
+impl Serialize for TokenId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        format!("T{:04}", self.0).serialize(serializer)
```

Comment:
Just an idea that it could be nice to also encode the symbol here to make the instance files more readable. But not necessary for this PR.